File: src/Module/Team/TeamEntity.java
package Module.Team;

import Module.Employee.EmployeeEntity;
import Module.Employee.EmployeeModel;
import Module.Location.LocationEntity;
import Module.Location.LocationModel;

import java.io.Serializable;
import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
public class TeamEntity implements Serializable {
    public int id;
    public String name;
    public int locationId;
    public int leaderEmployeeId;
    public List<EmployeeEntity> employeeEntities;
    public LocationEntity locationEntity;
    public EmployeeEntity leaderEmployeeEntity;


    public TeamEntity() {
    }

    public TeamEntity(TeamModel TeamModel, Object... objects) {
        this.id = TeamModel.getId();
        this.name = TeamModel.getName();
        this.locationId = TeamModel.getLocationId();
        this.leaderEmployeeId = TeamModel.getLeaderEmployeeId();
        for (Object object : objects) {
            if (object instanceof LocationModel) {
                this.locationEntity = new LocationEntity((LocationModel) object);
            } else if (object instanceof EmployeeModel) {
                this.leaderEmployeeEntity = new EmployeeEntity((EmployeeModel) object);
            } else if (object instanceof Collection) {
                for (Object o : (Collection<Object>) object) {
                    if (o instanceof EmployeeModel) {
                        this.employeeEntities = ((Collection<EmployeeModel>) object).parallelStream().map(EmployeeEntity::new).collect(Collectors.toList());
                        break;
                    }
                }
            }
        }
    }

    public TeamModel toModel() {
        TeamModel TeamModel = new TeamModel();
        TeamModel.setId(id);
        TeamModel.setLocationId(locationId);
        TeamModel.setName(name);
        TeamModel.setLeaderEmployeeId(leaderEmployeeId);
        return TeamModel;
    }
}

File: web/Angular4/src/app/Modules/Team/Team.Entity.ts
import {EmployeeEntity} from "../Employee/Employee.Entity";
import {LocationEntity} from "../Location/Location.Entity";

export class TeamEntity {

    id: number;
    name: string;
    locationId: number;
    leaderEmployeeId : number;
    employeeEntities: EmployeeEntity[];
    locationEntity: LocationEntity;
    leaderEmployeeEntity: EmployeeEntity;

    IsEdit: boolean;
    IsActive: boolean = false;
    IsSelected: boolean = false;
    constructor(data: any = null) {
        if (data == null) {
            this.id = null;
            this.name = null;
            this.locationId = null;
            this.leaderEmployeeId = null;
            this.leaderEmployeeEntity = new EmployeeEntity();
            this.locationEntity = new LocationEntity();
            this.employeeEntities = [];
        } else {
            this.id = data.id;
            this.name = data.name;
            this.locationId = data.locationId;
            this.leaderEmployeeId = data.leaderEmployeeId;
            if (data.employeeEntities != null) {
                this.employeeEntities = [];
                for (let item of data.employeeEntities) {
                    this.employeeEntities.push(new EmployeeEntity(item));
                }
            } else this.employeeEntities = [];

            if (this.locationEntity == null) this.locationEntity = new LocationEntity();
            if (this.leaderEmployeeEntity == null) this.leaderEmployeeEntity = new EmployeeEntity();
            if (this.employeeEntities == null) this.employeeEntities = [];
        }
        this.IsEdit = false;
    }
}